clar2wasm/src/error_mapping.rs

```diff
@@ -257,6 +269,70 @@ fn from_runtime_error_code(
         ErrorMap::ShortReturnExpectedValue => {
             let clarity_val = short_return_value(&instance, &mut store, epoch_id, clarity_version);
             Error::ShortReturn(ShortReturnType::ExpectedValue(clarity_val))
+
+        }
+        ErrorMap::ArgumentCountMismatch => {
+            let runtime_error_arg_offset =
+                get_global_i32(&instance, &mut store, "runtime-error-arg-offset");
+            let runtime_error_arg_len =
+                get_global_i32(&instance, &mut store, "runtime-error-arg-len");
+
+            let memory = instance
+                .get_memory(&mut store, "memory")
+                .unwrap_or_else(|| panic!("Could not find wasm instance memory"));
+            let arg_lengths = read_identifier_from_wasm(
+                memory,
+                &mut store,
+                runtime_error_arg_offset,
+                runtime_error_arg_len,
+            )
+            .unwrap_or_else(|e| panic!("Could not recover arg_name: {e}"));
+
+            let (expected, got) = extract_expected_and_got(&arg_lengths);
```

Comment:
This code is being reused in other parts of the code. Could you please refactor it into a separate function? Check the work done in the `short_return_value` function and try to replicate the approach here.

Reply:
To get rid of the duplication here I created a function `get_runtime_error_arg_lengths` based on the `short_return_value` function.